deploy/charts/rook-ceph-cluster/templates/cephcluster.yaml

```diff
@@ -5,6 +5,13 @@ metadata:
   name: {{ default .Release.Namespace .Values.clusterName }}
 spec:
   monitoring:
-{{ toYaml .Values.monitoring | indent 4 }}
+    enabled: {{ .Values.monitoring.enabled | default false }}
+{{- if .Values.monitoring.externalMgrEndpoints }}
+    externalMgrEndpoints:
+{{ toYaml .Values.monitoring.externalMgrEndpoints | indent 6 }}
+{{- end }}
+{{- if .Values.monitoring.externalMgrPrometheusPort }}
+    externalMgrPrometheusPort: {{ toYaml .Values.monitoring.externalMgrPrometheusPort }}
+{{- end }}
```

Comment:
Do we need to enclose this in an if check for whether `.Values.monitoring` has content or not? i.e., will this cause an exception if `.Values.monitoring` is unspecified?

Reply:
I had manually tested a few permutations, but missed that one. Looks like the chart already had a bug in that case, and also would be hit with these changes, so I'll push a fix, thanks for catching!

Reply:
As `.Values.monitoring` is part of the Chart provided values, I think it can never be undefined (unless there is a way to undefine a property in Helm ?).

Reply:
Good point. I was just able to simulate a failure in local development where I modified the built-in values.yaml, but you're right that it's not possible to remove the setting in the released chart.